src/common/src/array/arrow/arrow_iceberg.rs

```diff
@@ -297,4 +293,30 @@ mod test {
         ) as ArrayRef;
         assert_eq!(&arrow_array, &expect_array);
     }
+
+    #[test]
+    fn decimal_with_large_scale() {
+        let array = DecimalArray::from_iter([
+            None,
+            Some(Decimal::NaN),
+            Some(Decimal::PositiveInf),
+            Some(Decimal::NegativeInf),
+            Some(Decimal::Normalized("123.4".parse().unwrap())),
+            Some(Decimal::Normalized("123.456".parse().unwrap())),
+        ]);
+        let ty = DataType::Decimal128(28, 10);
```

Comment:
This test will pass with old code

Changing this can make it panic with overflow
```suggestion
        let ty = DataType::Decimal128(28, 20);
```

BTW, I'm not sure whether after the fix, all possible data won't panic any more, although the allowed range becomes larger. 🤔 

cc @xiangjinwu

Reply:
Previously, `10_i32.pow(diff_scale as u32)` is easy to overflow, because it uses i32 as a type, With i128, I think in most case it should work.
```
> Math.pow(2, 32)
4294967296
> Math.pow(2, 128)
3.402823669209385e+38
```

Reply:
> I'm not sure whether after the fix, all possible data won't panic any more

Say RisingWave stores 1e20 and tries to write in Arrow `Decimal128(38, 20)`, it will definitely overflow.